src/test/fio/fio_ceph_messenger.cc

```diff
@@ -0,0 +1,730 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ *  CEPH messenger engine
+ *
+ *  FIO engine which uses ceph messenger as a transport.  See corresponding
+ *  FIO client and server jobs for details.
+ */
+
+#include "global/global_init.h"
+#include "msg/Messenger.h"
+#include "messages/MOSDOp.h"
+#include "messages/MOSDOpReply.h"
+#include "common/perf_counters.h"
+
+#include <fio.h>
+#include <flist.h>
+#include <optgroup.h>
+
+enum ceph_msgr_type {
+	CEPH_MSGR_TYPE_UNDEF,
+	CEPH_MSGR_TYPE_POSIX,
+	CEPH_MSGR_TYPE_DPDK,
+	CEPH_MSGR_TYPE_RDMA,
+};
+
+const char *ceph_msgr_types[] = { "undef", "async+posix",
+				  "async+dpdk", "async+rdma" };
+
+struct ceph_msgr_options {
+	struct thread_data *td__;
+	unsigned int is_receiver;
+	unsigned int is_single;
+	unsigned int port;
+	const char *hostname;
+	const char *conffile;
+	enum ceph_msgr_type ms_type;
+};
+
+class FioDispatcher;
+
+struct ceph_msgr_data {
+	ceph_msgr_data(struct ceph_msgr_options *o_) :
```

Comment:
What is inline initializing? Since I consider this is a C implementation (with C++ mix) would be better to use gcc designated inits for plain old structrs (not for for msgr_data, since this has c++ members), i.e. struct point p = { .y = yvalue, .x = xvalue };, but old gcc does not support it so had to workaround.